rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/utils/DiskUtils.java

```diff
@@ -111,7 +110,14 @@ public static long getDirectorySizeInMB(File dirname)
     catch (Exception e) {
       log.error(String.format("Exception while calculating the size of the folder %s with exception : %s", dirname.toString(), e.toString()));
     }
-    return NumberUtils.toLong(output.toString().split("\\s+")[0]) / 1024;
+    try {
+      return Long.parseLong(output.toString().split("\\s+")[0]) / 1024;
+    }
+    catch (NumberFormatException e)
+    {
+      log.warn(String.format("Unable to calculate size of directory %s, du output returned: %s", dirname, output.toString()));
+      return -1L;
```

Comment:
In the library code the default value returned was 0.

Reply:
Yes, returning -1 to let the stats also have -1 (had missed the change in caller to forward -1) is better then returning 0 which implies dashboards on cache usage will show wrong data. With this we will know something is wrong in the system looking at stats/dashboards